tests/test_generate.py

```diff
@@ -23,12 +40,127 @@ def test_generator_get_screen_dimensions(generator):
     assert x == 120
     assert y == 205
 
+    test_embedded_screen = "tests/test_files/motor_bad.bob"
+    x, y = generator._get_screen_dimensions(test_embedded_screen)
+    assert x == 100
+    assert y == 100
+
+
+def test_generator_get_widget_dimensions(generator):
+    widget = Path("tests/test_files/widget.xml")
+
+    with open(widget) as f:
+        xml_content = f.read()
+
+    height, width = generator._get_widget_dimensions(xml_content)
+    assert height == 800
+    assert width == 1280
+
+    widget_bad = Path("tests/test_files/widget_bad.xml")
```

Comment:
Can you make this a separate test? Currently this test is testing a good screen and a bad screen, and they should be separate.
E.g.
```suggestion
def test_generator_get_widget_dimensions_bad_widget(generator):
    widget_bad = Path("tests/test_files/widget_bad.xml")
```